### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Julienexe**

### CODE REVIEW

File: firebase_db.py [hunk 1]
```diff
@@ -1,270 +1,68 @@
-import firebase_admin
```
Comment (Mikael967) [inline firebase_db.py:1]: Code Review for FirebaseDB Class
Overall Impression:
The code is well-structured and follows the Facade pattern effectively to simplify interactions with Firestore. The separation of concerns between UserRepository and NoteRepository is clear, and the methods are logically organized. However, there are a few areas where improvements can be made for readability, error handling, and consistency.

Specific Feedback
1. Code Readability and Documentation
Strengths:

The class and method docstrings are clear and provide a good overview of the functionality.

The use of type hints (str, Dict, List, etc.) improves code readability and helps with static analysis.

Suggestions:

Consider adding a brief docstring to the __init__ method to explain the purpose of initializing UserRepository and NoteRepository.

Add inline comments for complex logic, such as the rollback mechanism in add_note.

2. Error Handling
Strengths:

The rollback mechanism in add_note is a good practice to ensure data consistency.

Suggestions:

In delete_note, the note_ref is fetched directly from the database, but there’s no validation to ensure the note belongs to the user. This could lead to security issues. Consider adding a check similar to the one in edit_note.

In edit_note, the note_data check is good, but consider returning a more descriptive error (e.g., "Note not found" or "Unauthorized access") instead of just False.

Add error handling for database operations (e.g., create_note, delete_note) to handle cases where the database might be unavailable or the operation fails unexpectedly.

3. Consistency
Strengths:

The method names are clear and follow a consistent naming convention.

Suggestions:

In delete_note, the note_ref is fetched using db.collection("notes").document(note_id), which bypasses the NoteRepository. This inconsistency could lead to maintenance issues. Consider using self.note_repo for all note-related operations.

The font_family parameter in add_note and edit_note has a default value of "Arial". Ensure this default value is documented and consistent across the codebase.

4. Performance
Strengths:

The code avoids unnecessary database calls by fetching note_data only once in edit_note.

Suggestions:

In get_categories and get_notes, consider adding pagination or limits if the number of categories or notes could be large. This will improve performance and reduce memory usage.

5. Testing
Strengths:

The methods are modular and should be relatively easy to test.

Suggestions:

Ensure unit tests cover edge cases, such as:

Adding a note to a non-existent category.

Editing or deleting a note that doesn’t belong to the user.

Database failures during add_note or delete_note.

6. Security
Strengths:

The edit_note method includes a check to ensure the note belongs to the user.

Suggestions:

Add similar ownership checks in delete_note to prevent unauthorized deletions.

Consider validating user input (e.g., user_id, note_text, category) to prevent injection attacks or invalid data.

**Review by Mikael967 — COMMENTED** [review]
Great work overall! The code is clean and well-organized, and with these improvements, it will be even more robust and maintainable.
